PowerEditor/src/Notepad_plus.cpp

```diff
@@ -2086,18 +2086,19 @@ void Notepad_plus::checkLangsMenu(int id) const
 			if (curBuf->isUserDefineLangExt())
 			{
 				const TCHAR *userLangName = curBuf->getUserDefineLangName();
-				const int nbChar = 16;
-				TCHAR menuLangName[nbChar];
+				TCHAR menuLangName[langNameLenMax];
 
 				for (int i = IDM_LANG_USER + 1 ; i <= IDM_LANG_USER_LIMIT ; ++i)
 				{
-					if (::GetMenuString(_mainMenuHandle, i, menuLangName, nbChar-1, MF_BYCOMMAND))
+					if (::GetMenuString(_mainMenuHandle, i, menuLangName, langNameLenMax, MF_BYCOMMAND))

```

Comment:
See https://docs.microsoft.com/en-us/windows/desktop/api/winuser/nf-winuser-getmenustringa

> Note  The GetMenuString function has been superseded. Use the GetMenuItemInfo function to retrieve the menu item text.

But the really relevant issue is:

> Remarks
> 
> The nMaxCount parameter must be one larger than the number of characters in the text string to accommodate the terminating null character.
> 
> If nMaxCount is 0, the function returns the length of the menu string.
> Security Warning
> The lpString parameter is a TCHAR buffer, and nMaxCount is the length of the menu string in characters. Sizing these parameters incorrectly can cause truncation of the string, leading to possible loss of data.

, so the necessary string buffer length could be retrieved dynamically, if wanted.
I think also langNameLenMax still needs the -1 as before.

Reply:
Hmmm... Same link says: (also during debug below points stand true). Sometime MS documentation is confusing...

>Type: LPTSTR
The buffer that receives the null-terminated string. **If the string is as long or longer than lpString, the string is truncated and the terminating null character is added**. If lpString is NULL, the function returns the length of the menu string.
cchMax
Type: int
The maximum length, in characters, of the string to be copied. **If the string is longer than the maximum specified in the nMaxCount parameter, the extra characters are truncated**. If nMaxCount is 0, the function returns the length of the menu string.

Look from "GPD GPD GPD GPD GPD GPD GPD GPD GPD GPD GPD GPD GPD GPD GPD 1122" last character is truncated and null termination is added. If we use -1 in length then we are going to lose one more character which means.... New length will be 62 characters instead of 63. 😊 

![image](https://user-images.githubusercontent.com/14791461/55017282-68d80180-5016-11e9-87c9-4b043be733c6.png)

Reply:
@chcg just for reference. look at the screenshot when -1 is used..

![image](https://user-images.githubusercontent.com/14791461/55018024-03851000-5018-11e9-95f6-63cacaff747c.png)

Reply:
@SinghRajenM Thanks for debugging this. So if the terminating null is there also in case of truncation everything is fine.